lib/std/io/reader.zig

```diff
@@ -66,7 +66,7 @@ pub fn Reader(
             while (true) {
                 array_list.expandToCapacity();
                 const dest_slice = array_list.items[start_index..];
-                const bytes_read = try self.readAll(dest_slice);
+                const bytes_read = try self.read(dest_slice);
```

Comment:
This change breaks other behavior in the function.

`readAll()` returns some value other than `dest_slice.len` when the end of stream is reached, while `read()` returns 0 when the end of stream is reached.

Down below in line 77 there is a check for `bytes_read != dest_slice.len` aka *end of stream reached*. Using `read()` this has to be changed to `bytes_read == 0`.

Reply:
thanks for review!
in my opinion, .readAll and .readAllAlloc both employ a while loop on .read, and .readAll's return has no difference with .read's if you looked their impl.